### DISCUSSION TITLE
0.97.0.0 for MC 1.14.* & 1.15.* & 1.16.*

### DISCUSSION THREAD

**Discussion by LlmDl** [TownyAdvanced/Towny#4922] [Release Discussions]
# Towny Advanced 0.97.0.0

<p><img align=left style="padding: 40px;" src="https://user-images.githubusercontent.com/879756/65964696-19d6b300-e423-11e9-9cb0-d193225ee40f.png">

### Release contains:
>Towny 0.97.0.0
>TownyChat 0.88

|Welcome to Towny 0.97.0.0.|
| ---- | 
| Towny 0.97.0.0 is the first release in the 0.97.\*.\* series. It brings about an expanded API and superior safe-mode prevention through a re-organized database compared to older versions. Of course there are plenty of new features as well. |
| Included in this release (compared with 0.96.7.0) are some cool new features like Town merging, command-blacklisting by location, custom metadata and some smaller suggestions made on the issue tracker. |
| We also have a couple of critical bug fixes: if you are using any of these versions of Minecraft: 1.14.\*/1.15.\*/1.16.\*, you need to update to this version of Towny as soon as possible. |
| This release also drops the API deprecated in 2019 from Towny's source code. Most plugins that were still using this API have already replaced their Towny support with the modern API. |
| If you're upgrading from 0.96.2.0, 0.96.3.0, 0.96.4.0, 0.96.5.0, 0.96.6.0 or 0.96.7.0 there are 0 REQUIRED changes. This is a drag-and-drop update: just replace your old .jars with the new ones and Towny will take care of the rest of the update by itself. |

<details>
<summary> ➕ Added</summary><blockquote>

## ➕ Added:
  - New Feature: Visualized SpawnPoints
    - Adds a ring of particles on top of the various spawn points.
    - Includes Jail, Outpost, Town and Nation Spawns.
    - Important for the coming Jail Overhaul.
  - Made GUI titles translatable.
    - Closes #4719.
  - Make it so that admins can use /plot evict on any resident-owned plot.
    - Closes #4678.
  - Change it so that admins with the bypass node will not warn towns that an outlawed admin is spying on them.
    - Previously only full admins were preventing warns.
  - Add ability to set town and nation tags automatically using the town/nation name.
    - Closes #4869.
  - Add ability to make capital cities have a higher resident cap.
    - Closes #4841.
  - Add option to require FrostWalker enchantment to require build rights to freeze water.
    - Closes #4887
  - Add config option to make plots lost from non-payment of plottaxes go up for sale.
    - Closes #4892.
  - Allow a bankrupt town with no townblocks claimed to claim one plot.
    - Closes #4452. 
  - Add towny.command.plot.perm.hud to the nomad group in townyperms.yml by default.
  - Update banned town and nation names to reflect some of the newer commands.
  - Do a tidy of the PlayerCacheUtil so we aren't throwing and catching NREs.
  - /ta nation NATIONNAME add TOWNNAME now bypasses the invite and other rules allowing an admin to force-add a town to a nation.
  - Inn Plots no longer turn switch perms on for all groups, instead it uses the TownBlockOwner's perm line like most plot types.
  - Small refactor of the bank balance caching. 
  - Add missing towny.command.town.reclaim to default townyperms.yml.
</blockquote>
<blockquote>

  - Sort town resident list by date they joined it, courtesy of Warriorrrr with PR #4880.
    - Also sorts Towns by join date in the Nation,
    - Also displays claim time in /plot perm.
    - Closes #4556.
  - Town Merging, courtesy of Warriorrrr with PR #4755.
    - Closes #3193.
  - Add config option for toggling visualized spawn points, courtesy of Warriorrrr with PR #4865
    - Closes #4857.
  - Add town/nation name blacklist config option, courtesy of Warriorrrr with PR #4878.
    - Closes #4875.
  - Add ability to show chunk notifications in the BossBar, courtesy of Warriorrrr with PR #4888.
    - Closes #4882
  - Customizable tag lengths, courtesy of Warriorrrr with PR #4907.
    - Closes #4904.
  - Add auto town name capitalisation, courtesy of Warrior with PR #4898.
    - Closes #4891.
  - Small HUDManager overhaul, courtesy of Warriorrrr with PR #4726.
  - Add townyadmin command for giving/removing nation ranks, courtesy of Warriorrrr with PR #4736.
  - Add config option for ignoring min_distance's for allies, courtesy of Warriorrrr with PR #4752.
  - Add townyadmin command for adding/removing nation allies or enemies, courtesy of Warriorrrr with PR #4826
  - Add Adventure library, courtesy of Warriorrrr with PR #4870.
</blockquote>
<blockquote>

  - Update POM: Dependencies, Plugins, and HTTPS, courtesy of TheFlagCourier with PR #4749.
  - Add localization strings for Towny's FlatFile datasource handler, courtesy of TheFlagCourier with PR #4761</blockquote>
<blockquote>

  - Custom Metadata Types, courtesy of silverwolfg11 with PR #4700.
</blockquote>
<blockquote>

  - New command blocking, courtesy of EuGNosii with PR #4849.
    - Allows blocking commands inside towns and limiting them to plots owned by the players only.
    - Useful for limiting sethome/home commands to plots owned by the players themselves and not someone else.
    - Admins and players with the towny.admin.town_commands.blacklist_bypass permission node will not be hindered.
    - Closes #4835.
</blockquote>

----
</blockquote></details>

<details>
<summary>➕ Automatic Config Edits (0)</summary><blockquote>

### ➕ Towny will automatically edit the following settings in your config and existing worlds
  - None
  
----
</blockquote></details>

<details>
<summary>✨ Command Changes (7)</summary><blockquote>

<details>
<summary>✨ New Commands (7)</summary>

## ✨ New Commands:
  - /t merge town
    - town is the town which will be merged into the town owned by the mayor using the command.
    - Comes with potential economic costs (related to base merging costs, as well as townblock costs.)
  - /ta toggle wildernessuse (on|off)
    - Toggles build/destroy/switch/itemuse in the wilderness of each world off or on.
    - Toggles on if on/off is not specified.
  - /ta toggle regenerations (on|off)
    - Toggles explosion regen and unclaimed revert in the wilderness of each world off or on.
    - Toggles off if on/off is not specified.
  - /ta database remove titles
    - Removes all titles and surnames from all residents.
  - /ta town TOWNNAME unruin
    - Closes #4824.
  - /ta nation NATIONNAME ally [add|remove] NATIONNAME
  - /ta nation NATIONNAME enemy [add|remove] NATIONNAME

----
</details>

<details>
<summary>❌ Removed Commands (0)</summary>

## ❌ Removed Commands:
  - None

----
</details>

----
</blockquote></details>

<details>
<summary>📕 Config Changes (19)</summary><blockquote>

<details>
<summary> 📕 New Config Options (19)</summary>

#### These new settings will be added to your config with their default settings.
  - global_town_settings.max_residents_capital_override
    - Default: 0
    - The maximum number of residents that can be joined to a capital city.
    - Requires max_residents_capital_override to be above 0.
    - Uses the greater of max_residents_capital_override and max_residents_per_town.
  - global_town_settings.town_command_blacklisting.enabled
    - Default: false
    - Allows blocking commands inside towns through the town_blacklisted_commands setting.
    - This boolean allows you to disable this feature altogether if you don't need it.
  - global_town_settings.town_command_blacklisting.town_blacklisted_commands
    - Default: somecommandhere,othercommandhere
    - Comma separated list of commands which cannot be run inside of any town.
  - global_town_settings.town_command_blacklisting.player_owned_plot_limited_commands
    - Default : sethome,home
    - This allows the usage of blacklisted commands only in plots personally-owned by the player.
    - Players with the towny.claimed.townowned.* permission node will be able to run these commands inside of town-owned land. This would include mayors, assistants and possibly a builder rank.
    - Players with the towny.claimed.owntown.* permission node (given to mayors/assistants usually, will also not be limited by this command blacklist.
  - global_town_settings.prevent_fluid_griefing
    - Default: true
    - When enabled, blocks like lava or water will be unable to flow into other plots, if the owners aren't the same.
  - global_town_settings.automatic_capitalisation
    - Default: false
    - When enabled, town names will automatically be capitalised upon creation.
  - town.min_distances_ignored_for_towns_in_allied_nation
    - Default: false
    - If true, the settings: min_plot_distance_from_town_plot and min_distance_from_town_homeblock will be ignored for towns that are mutually allied. 
    - Setting to false will keep all towns separated the same if they are allies.
  - town.max_distance_for_merge
    - Default: 10
    - The maximum distance (in townblocks) that 2 town's homeblocks can be to be eligible for merging.
  - town.set_tag_automatically
    - Default: false
    - Setting this to true will set a town's tag automatically using the first four characters of the town's name.
  - town.default_taxes.plot_tax.does_non-payment_place_plot_for_sale
    - Default: false
    - Does a player's plot get put up for sale if they are unable to pay the plot tax?
    - When false the plot becomes town land and must be set up for-sale by town mayor or staff.
  - economy.new_expand.price_town_merge
    - Default: 0
    - The base cost a town has to pay to merge with another town. The town that initiates the merge pays the cost.
  - economy.new_expand.price_town_merge_per_plot_percentage
    - Default: 50
    - The percentage that a town has to pay per plot to merge with another town. The town that initiates the merge pays the cost.
    - This is based on the price_claim_townblock.
  - plugin.visualized_spawn_points_enabled
    - Default: true
    - If enabled, particles will appear around town, nation, outpost & jail spawns.
  - plugin.name_blacklist
    - Default: empty
    - A blacklist used for validating town/nation names.
    - Names must be seperated by a comma: name1,name2
  - invite_system.confirmation_timeout
    - default: 20
    - How many seconds before a confirmation times out for the receiver. 
    - This is used for cost-confirmations and confirming important decisions.
  - nation.set_tag_automatically
    - Default: false
    - Setting this to true will set a nation's tag automatically using the first four characters of the nation's name.
  - protection.prevent_frost_walker_freezing
    - Default: false
    - When set to true, players with the Frost Walker enchant will need to be able to build where they are attempting to freeze.
  - notification.notifications_appear_on_bossbar
    - Default: false
    - When enabled, notifications will appear on the bossbar instead of in chat or above the action bar.
    - Uses the duration from below for how long the bossbar appears.
  - filters_colour_chat.modify_chat.max_tag_length
    - Default: 4
    - Maximum length for Town and Nation tags.

----
</details>

<details>
<summary>❌ Removed Config Options (0)</summary>

## ❌ Removed Config Options:
  - none

----
</details>

----
</blockquote></details>

<details>
<summary>🔑 Permission Node Changes (4)</summary><blockquote>

<details>
<summary>🔑 New Permission Nodes (4)</summary>

## 🔑 New Permission Nodes:
  - towny.command.town.merge.
    - Child node of towny.command.town.*
  - towny.command.townyadmin.nation.enemy
  - towny.command.townyadmin.nation.ally
    - both are child nodes of towny.command.townyadmin.nation and therefore you will not need to add these (in most situations.)
  - towny.admin.town_commands.blacklist_bypass
    - Child Node of towny.admin
    - Allows admins to not be restricted by either command-blocking lists.

----
</details>

<details>
<summary>❌ Removed Permission Nodes (0)</summary>

## ❌ Removed Permission Nodes:
  - None

----
</details>

----
</blockquote></details>

<details>
<summary>💻 API Changes (28)</summary><blockquote>

## 💻 API: 
  - Added convenience methods to TownyAPI:
    - getNation(name)
    - getTown(name)
    - getResident(name)
    - getNation(uuid)
    - getTown(uuid)
    - getResident(uuid)
    - getResidentTownOrNull(resident)
    - getResidentNationOrNull(resident)
    - getTownNationOrNull(town)
  - New damage/pvp events:
    - TownBlockPVPTestEvent - thrown when a townblock in a town has the PVP status decided.
    - WildernessPVPTestEvent - thrown when determining what the wilderness status for PVP is.
    - TownyDamageEvent - Abstract event for the following:
    - TownyPlayerDamagePlayerEvent - used to override players hurting players.
    - TownyExplosionDamagesEntityEvent (Part of the earlier API expansion, now moved to the damage package and extending the TownyDamageEvent.)
  - TownyExplodingBlocksEvent 
    - Add getBukkitExplodeEvent for convenience.
  - Added ResidentPreJailEvent, a Cancellable event.
    - Closes #4862.
  - Removal of any methods deprecated earlier than 0.96.0.0.
    - Marked many deprecated methods with the version in which they were deprecated for easier future removal.
  - Deprecated: EconomyException
    - The economy exception is no longer thrown by Towny.
    - After close examination it was found that the exception is no longer needed.     

  - Add PlotToggle events to toggling plot groups, courtesy of Warriorrrr with PR #4748.
  - Add a TownKickEvent, courtesy of Warriorrrr with PR #4831.
  - Add NationKingChangeEvent and TownMayorChangeEvent, courtesy of Warriorrrr with PR #4834.
  - Add TownPreInvitePlayerEvent, courtesy of Warriorrrr with PR #4867.

  - Add 3 new events around the displayed nation list, courtesy of Goosius with PR #4810.
    - NationListDisplayedNumOnlinePlayersCalculationEvent
    - NationListDisplayedNumResidentsCalculationEvent
    - NationListDisplayedNumTownsCalculationEvent
  - Add event for the calculation of town map color. courtesy of Goosius with PR #4809.
    - TownMapColourCalculationEvent
    
  - Deprecate FlagWar Classes, courtesy of TheFlagCourier with PR #4920.
    - Is being replaced by the standalone FlagWar plugin.

----
</blockquote></details>

<details>
<summary>🔧 Fixed</summary><blockquote>
  
## 🔧 Fixed: 
  - Fix "/res toggle reset" not resetting modes.
    - Closes #4730.
  - Fix empty-selection message showing when claiming single plots is blocked because the area is actually already claimed.
    - Closes #4729.
  - Fix nation merging.
    - Closes #4732.
  - Fix /ta command for setting town title/surnames not respecting the admin override.
    - Closes #4741.
  - Fix event war plot attacking not working.
    - Closes #4740.
  - Fix being unable to claim a personally-owned plot being sold by a resident.
    - Closes #4767.
  - Adapt to spigot bug being closed, allowing us to prevent explosions from dropping items without setting the exploded block to air.
    - Closes #4770, #4771.
  - Fix up double-chest-on-plot-borders visual bugginess.
    - Closes #4783.
  - Fix kings who give up their position willingly, not losing their King permission nodes given in TownyPerms.yml
    - Closes #4784.
  - Fix the method of dealing with PVP damage caused by Channeling Tridents, also preventing hurting of mobs where PVP was disabled.
    - Closes #4778.
  - Fix for NPE when someone activates another scoreboard (ex: featherboard) while the perm hud is active.
    - Closes #4754. 
  - Fix "Error Updating item_use/build/destroy permission" messages seen in EventWar.
    - Closes #4802.
  - Fix Spawn Point particles not disappearing when spawn points are moved.
    - Closes #4830.
  - Fix unclaimedZoneIgnoreIds not working properly.
    - Closes #4844.
  - Fix personally-owned plots not taking on a fully-open perm line when a town is ruined.
    - Closes #4846.
  - Fix plot notifications not showing information when you've left a plotgroup into a non-plotgroup townblock.
    - Closes #4856.
  - Fix sneaky Citizens/Sentinal NPCs being able to get through the tests surrounding death payments.
    - Closes #4895.
  - Fix NPCs being able to throw an exception when they kill a player.
  - Fix up the message shown to admins when a player has tried to log in with a dis-allowed name.
  - Fix plot names having their spaces parsed out.
  - Fix debug messaging in DailyTimerTask.
  - Fix error message being sent to resident instead of player.
  - Fix goofy replacing in TownySpigotMessaging sendSpigotConfirmMessage.
  - Fix regression in GUIs which made inventories unusable.
  - Fix BookFactory skipping the 15th line of a book.
  - Fix potential safe mode from removing corrupted nation which cannot return a king.
  - Make /t set outpost do more than show null when used in the wilderness.
  - Make toggling REDSTONE_WIRE into a destroy test.
  - Correctly show the proper already registered messages when using /ta set mayor TOWN PLAYER command.
  - Fix possible NPE when unjailing a resident.
  - Fix mayors unable to use /plot nfs on personally-owned plots owned by their town.
  - Fix PVP-related issue where people could be hit once in non-PVP areas.
  - Fix maxPercentTaxAmount town setting not being persistent.
  - Fix /res tax not accounting for the maxPercentTaxAmount town setting.
  - Potential fix for PermHud being able to throw an NPE on line 86.
  - Fix $0 plot taxes being paid by residents.
  - Fix for a "null" feedback message when using /t outlaw add NAME.
  - Fix EventWar not turning PVP on in the townblocks where the war is active.
  - Fix TownPreMergeEvent being set improperly.
  - Fix NPE when deleting a Town.
  - Fix unbankrupting a town resetting a town's bank account, vanishing any money that might have been added to the town bank manually during bankruptcy.
  - Fix ConcurrentModificationException in Town.
  - Potential fix for being show "Not registered." when attempting to rename a town.
  - Fix TownyWorld setting a town's world when a town is added to a townyworld.
  - Fix HangingEntity placement all using Painting material for permission tests.
  - Fix not using the filteredname earlier on when a town is made. 
  - Lower CreatureSpawnEvent priority to make Towny play nice with EliteMobs and not be the worst.
  - Fix missing confirmation for admins using /t delete NAME and /n delete NAME.
  - Fix Economy formatting with colours able to muck about with the bankhistory books.
  - Make the spawn particles chill out.
  - Fix spawn particles not disappearing when a town or nation is deleted.
  - Fix plotgroups being able to be named with characters that ought to be filtered out.
  - Make plotgroups which load and throw an AlreadyRegisteredException actually print out which one is no good.
  - Fix missing tab completions for /towny map big & /town outpost list.
  - Fix potential NPE in revert-on-unclaim feature.
  - Rework negative upkeep to probably make it compatible with more economy plugins.
  - Fix it so that servers who remove the numResidents: 0 town and nation levels will not generate NPE.
  - Fix SQL_Schema that lead to errors on 0.96.7.15 MySQL servers. 
  - Fix potential NPE when loading a resident.
  - Fix missing permission node test.
</blockquote>
<blockquote>
 
  - Fix free town/nation rename exploit, courtesy of Warriorrrr with PR #4743.
  - Remove 1.14 crafting blocks from config's switch ids, courtesy of Warriorrrr with PR #4744.
  - Fix town delete message being sent twice, courtesy of Warriorrrr with PR #4746.
  - Fix empty nations not being deleted on startup with flatfile, courtesy of Warriorrrr with PR #4753.
  - Fix formatted balance for not enough money error, courtesy of Warriorrrr with PR #4833.
  - Misc fixes courtesy of Warriorrrr with PR #4854.
  - Fix entity-vs-entity combat always being allowed, courtesy of Warriorrrr with PR #4893.
  - Fix water & lava being usable to grief others, courtesy of Warriorrrr with PR #4855.
  - Fix german.yml FlagWar message, courtesy of Warriorrrr with PR #4899.
  - Fix /n list & /t list by online showing total residents, courtesy of Warrorior with PR #4901.
</blockquote>
<blockquote>

  - Fix for displayed-nation-list events, courtesy of Goosius with PR #4814.
</blockquote>

----    
</blockquote></details>

<details>
<summary>🌏 Language Updates</summary><blockquote>

## 🌏 Language Updates:
  - Bump language files to 0.105.
  - Update spanish.yml, courtesy of ElMoha943 with PR #4716, #4722.
  - Update es-419.yml, courtesy of EuGNosii with PR #4717, #4723, #4787, #4848, #4881.
  - Update korean.yml, courtesy of yumjunstar with PR #4811, #4889, #4905
  - Swedish translation, added courtesy of OskyEdz with PR #4840, #4853.
  - Danish translation, courtesy of Davidoen.
  - French translation updated, courtesy of PainOchoco with PR #4864.

----
</blockquote></details>

<details>
<summary>👽 Placeholder Updates (6)</summary><blockquote>

## 👽 Placeholder Updates:
  - New Placeholders:
    - %townyadvanced_is_nation_peaceful%
    - %townyadvanced_is_town_peaceful%
      - If the player has a town/nation which is peaceful the language string "status_town_title_peaceful" will be shown: "&b(Peaceful)".
      - Addresses https://github.com/TownyAdvanced/SiegeWar/issues/72
    - %townyadvanced_player_location_plot_name%
      - Displays a plot's name if it has one.

  - New RELATIONAL placeholder: %rel_townyadvanced_color%
    - Applies the colour codes set in the config.yml's new papi_relational_formatting section.
    - Used in TAB and other plugins which can do Relational placeholders.
    - Closes #4747, Closes #4349.

----
</blockquote></details>

<details font-size:20>
<summary>💬 TownyChat Changes</summary><blockquote>
  
## 💬 TownyChat Changes: 

  - Fix firing PlayerJoinChatChannelEvent async.
  - Fix event firing async.
  - Fix ranged global channels' chat being forwarded to dynmap.
  - Fix UnsupportedOperationException which was possible when leaving a channel.
  - Fix players who've just recently joined a town/nation/alliance not being included in the town/nation/alliance chats.
  - Fix spam_time settings not working.
  - Fix hex color codes not being parsed in TownyChat spy messages. 

----
</blockquote></details>

<table align=center>
<tr><th colspan=4><h3>Towny Sponsors<br><h4>I want to give a big thank you to all of my sponsors:<br>These are the people that help to make Towny's support and development as active as it is.</th></tr>
<tr><td>MrLogano</td><td>Momshroom</td><td>Cobrex1</td><td> KarlOfDuty </td></tr>
<tr><td>scatterspell</td><td>gerardvanschip</td><td>partydev</td><td> DrprofLuigi </td></tr>
<tr><td>Tobza </td><td>SmallSansSerif </td><td>TheNylox</td><td>diontimmer </td></tr>
<tr><td>trevor1097</td><td>benevolent2012 </td><td> bretomart </td><td>HunSolos</td></tr>
<tr><td>ofunny</td><td>0xBit-MC</td><td>specs1396</td><td>GoldyOne</td></tr>
<tr><td>treeDerevo</td><td>Scottlander100</td><td>teozfrank</td><td>ItsSynchro</td></tr>
<tr><td>Advanced-Kind-MC</td><td>MidnightSugar</td><td>ceeedric</td><td>0liverGMC</td></tr>
<tr><td>Microhazard</td><td>mhiaxo</td><td colspan=2>and 2 private sponsors.</td></tr>
<tr><td colspan=4 align=center><h3><a href=https://github.com/sponsors/LlmDl>If you want to support the developer consider becoming a sponsor.</a><br><h6>(It's just like Patreon but 100% goes to the developer.)</td></tr>
</table>

<table align=center>
<tr><th colspan=3> Important Links </th><tr>
<tr align=center><td><a title="How Towny Works" href="https://github.com/TownyAdvanced/Towny/wiki/How-Towny-Works">How Towny 
Works</a></td><td><a title="Install Guide" href="https://github.com/TownyAdvanced/Towny/wiki/Installation">Towny Install Guide</a></td><td><a title="Updating Towny" href="https://github.com/TownyAdvanced/Towny/wiki/Updating-Towny">Towny Update Guide</a></td>
</tr>
<tr align=center>
<td><a title="Towny Changelog" href="https://raw.githubusercontent.com/TownyAdvanced/Towny/master/resources/ChangeLog.txt" target="_blank" rel="noopener">Complete Changelog</a> </td><td><a title="Default Config Files" href="https://github.com/TownyAdvanced/Towny/wiki/Config-Files">Default Config Files</a></td><td><a title="Commands/Permissions/Placeholders" href="https://github.com/TownyAdvanced/Towny/wiki/Reference">Commands/Permissions<br>Placeholders</a></td>
</tr>
</table>

### ⏬ Download available as a .zip file in the Assets section below:

<hr /><em>This discussion was created from the release <a href='https://github.com/TownyAdvanced/Towny/releases/tag/0.97.0.0'>0.97.0.0 for MC 1.14.* & 1.15.* & 1.16.*</a>.</em>

**Comment by ggvbo** [comment]
Nice!

**Comment by MilesBHuff** [comment]
@LlmDl 
> If you're upgrading from 0.96.2.0, 0.96.3.0, 0.96.4.0, 0.96.5.0, 0.96.6.0 or 0.96.7.0 there are 0 REQUIRED changes.

Does this apply to point releases?  We're on 0.96.2.12, and I'm in the process of updating prerelease-by-prerelease until 0.96.3.0, at which point I was planning to update release-by-release until we reached the latest version.

In addition to the above question, I just want some quick clarification:  Do you *really* mean that there are no manual changes required between 0.97.0.0 and 0.96.3.0?  How is that possible?

Thanks

**Reply by LlmDl** [reply]
You can jump from 0.96.2.12 all the way to 0.97.0.0. I *really* mean there are no manual changes for anyone on 0.96.2.0 or newer (even the pre-releases after 0.96.2.0.)

We have a config-migrator that will add the things you'd normally have to add manually, which is why it is possible.

You should not have let your server fall so far behind on updates. There are many major flaws in older versions.

**Reply by MilesBHuff** [reply]
Crazy!  Okay.  I will experiment with doing just that, then.  Thanks!

And I know.  Normally I try to keep on-top of things.  But Dell bricked my computer during a repair visit, and it took them over a month to send me a replacement.  That was immediately followed by Winter holidays (when I was out of town), and then moving to a new house.  I finally got a good local dev environment set-up two months ago, and have gradually been getting plugins updated.  Towny was last in-line because of how daunting the process was.

**Reply by LlmDl** [reply]
Sounds like a perfect storm. You'll be happy to know that the update process is probably the easiest of any plugin there is: just drag-and-drop unless otherwise stated.

We automatically add new config values at their stated default settings.
We automatically edit existing config values with new blocks/things as needed.

**Reply by MilesBHuff** [reply]
Just finished, and I have to say -- that *was* fairly painless!  I did version-by-version until 0.96.3.0, which I'm glad I did, because the documentation changes would have made the diff from 0.96.3.0 to 0.97.0.0 quite terrible, lol.
Thanks again for all the hard work!

Unrelated:  I'm planning to become a regular patron if the server ever turns enough of a profit;  but at present, it's barely breaking even.
